decompiler/util/widget.py

```diff
@@ -247,101 +248,100 @@ def reanalyze(self):
-        """Update visibility of widget"""
-        self._is_widget_visible = is_visible
-
     @Decorators.requires_function
     def update_code_view(self):
         """Reload CodeDisplay content from cache"""
         self.editor.set_font()  # responsive gui settings
         self.editor.setPlainText(self._cache.get(self._current_function.name, ""))
         self.highlighter.rehighlight()
 
-    def update_cache(self, result: Tuple[str, str]) -> None:
+    def callback_update_cache(self, result: Tuple[str, str]) -> None:
         """Callback for successful decompilation"""
         function_name, code = result
         self._cache[function_name] = code
 
-    def decompilation_error(self, error: Tuple):
+    def callback_decompilation_error(self, error: Tuple):
         """Callback for decompilation error"""
         task_name, *error_messages = error
         message = f"Decompilation of `{task_name}` failed with:\n\n" + "\n\n".join(str(e) for e in error_messages)
         self.editor.setPlainText(message)
         self._cache[task_name] = message
 
-    def worker_task_started(self, task_name: str):
+    def callback_worker_task_started(self, task_name: str):
         """Callback for decompilation task started"""
         logging.info(f"Started decompilation of {task_name}")
         self._cache[task_name] = "In Queue"
         self.update_code_view()
 
-    def decompilation_finished(self):
+    def callback_decompilation_finished(self):
         """Callback for finished decompilation (updates code view)"""
         logging.info("Finished decompilation")
         self.update_code_view()
 
     def start_worker(self, binary_view: BinaryView, function: Function) -> None:
         """Set up Worker with callbacks"""
         worker = Worker(binary_view, function)
-        worker.signals.result.connect(self.update_cache)
-        worker.signals.task_name.connect(self.worker_task_started)
-        worker.signals.error.connect(self.decompilation_error)
-        worker.signals.finished.connect(self.decompilation_finished)
+        worker.signals.result.connect(self.callback_update_cache)
+        worker.signals.task_name.connect(self.callback_worker_task_started)
+        worker.signals.error.connect(self.callback_decompilation_error)
+        worker.signals.finished.connect(self.callback_decompilation_finished)
         self.threadpool.start(worker)
 
-    @classmethod
-    def create_widget(cls, name: str, parent: ViewFrame, data=None):
-        """Called by Qt method addDockWidget to create widget"""
-        return cls(parent, name)
+    def updateState(self):
+        # Get the currently active view frame for this group of panes. There can be
+        # multiple view frames in a single window, or the pane could be popped out
+        # into its own window. UIContext.currentViewFrameForWidget will determine
+        # the best view frame to use for context.
+        self._current_frame = UIContext.currentViewFrameForWidget(self)
```

Comment:
pretty verbose - maybe rahter hab a `get_current_frame` function with a nice DocString if you deem `UIContext.currentViewFrameForWidget(self)` hard to comprehend ;)

Reply:
`UIContext.currentViewFrameForWidget(self)` comes from BN API. Could add a wrapper, but we only need it here once